gammapy/estimators/points/sed.py

```diff
@@ -136,13 +136,31 @@ def run(self, datasets):
         if not datasets.energy_axes_are_aligned:
             raise ValueError("All datasets must have aligned energy axes.")
 
-        if "TELESCOP" in datasets.meta_table.colnames:
-            telescopes = datasets.meta_table["TELESCOP"]
-            if not len(np.unique(telescopes)) == 1:
-                raise ValueError(
-                    "All datasets must use the same value of the"
-                    " 'TELESCOP' meta keyword."
-                )
+        if (
+            len(datasets) > 1
+            and datasets[0].meta_table is not None
+            and any(
+                dataset.meta_table["TELESCOP"].shape
+                != datasets[0].meta_table["TELESCOP"].shape
+                for dataset in datasets
+            )
+        ):
+            for dataset in datasets:
+                if "TELESCOP" in dataset.meta_table.colnames:
```

Comment:
This is testing that every dataset has only one single TELESCOP value, not that these values are always the same.

Reply:
modified and simplified